www/guides/react.md

```diff
@@ -0,0 +1,373 @@
+
+Prerequisites: Snowpack is a command line tool installed from npm. This guide assumes a basic understanding of Node.js, npm, and how to run commands in the terminal. Knowledge of React is not required, Snowpack is a great way to learn React.
+
+## Getting started with Snowpack and React
+
+In this step by step you'll learn how to use the [`create-snowpack-app`](https://www.snowpack.dev/#create-snowpack-app-(csa) tool to spin up a working Snowpack project and then how to add React to it.
+
+`create-snowpack-app` is a command line tool that creates a working Snowpack App based on a template. The template you're starting with here is `@snowpack/project-template-minimal is` a bare-bones setup where Snowpack serves an index.html, index.css, and index.js.
+
+```bash
+npx create-snowpack-project react-snowpack --template @snowpack/project-template-minimal
+cd react-snowpack
+npm run start
+```
+
+Image: screenshot of project-template-minimal, which shows "Hello world" in text on a white background
+
+This is about as basic as it gets. When you run `npm start` Snowpack serves an `index.html`, `index.css`, and `index.js`.
+
+> The `README.md` contains useful information about what each file does.
+
+The first thing that you'll want to do is install React in your project. To do that, run the following command in your project directory:
+
+To install these you can run:
+
+```bash
+npm install react react-dom --save
+```
+
+> In this tutorial commands use `--save` for frontend libraries and `--save-dev` for your developer tools to keep them organized separately. This is totally up to you, though.
+
+## Create your first React component
+
+React uses a syntax called [JSX](https://reactjs.org/docs/introducing-jsx.html) for components. Snowpack has built in support for JSX files with the extension `.jsx`. To create your first React component in Snowpack rename `index.js` to `index.jsx`.
+
+> You may have used JSX in `.js` files before, but Snowpack expects JSX to be .jsx files
+
+```bash
+mv index.js index.jsx
+```
+
+> Remember that your index.html file is importing `/index.js`, not `/index.js`. Do you need to update it to match the new file? You don't because Snowpack turns `index.jsx` (which wouldn't run in the browser) into the `index.js` (which would).
+
+You can now import React into `index.jsx` and render a simple test component just to make sure it's working.
+
+Import React and add this test component `index.jsx` so you can use it.
+
+```diff
+/* Add JavaScript code here! */
+- console.log('Hello World! You did! Welcome to Snowpack :D');
++import React from 'react';
++import ReactDOM from 'react-dom';
++ReactDOM.render(<div>"HELLO REACT"</div>, document.getElementById('root'));
+```
+
+If you run this code you'll get `Error: Target container is not a DOM element.` The call to `document.getElementById('root')` is failing because the HTML page has no element with that ID. Head to index.html and add one in the `<body>`
```

Comment:
several minor thigns:
- `.` in backticks for error
- `index.html` not being in code ticks despite files being in code ticks everywhere else
- comma after "if you run this code" feels natural

fixed:
```
If you run this code, you'll get `Error: Target container is not a DOM element`. The call to `document.getElementById('root')` is failing because the HTML page has no element with that ID. Head to `index.html` and add one in the `<body>`.
```